src/main/java/com/iexec/sms/secret/Secret.java

```diff
@@ -17,63 +17,27 @@
 package com.iexec.sms.secret;
 
 import lombok.AccessLevel;
-import lombok.AllArgsConstructor;
 import lombok.Getter;
 import lombok.NoArgsConstructor;
-import org.hibernate.annotations.GenericGenerator;
 
 import javax.persistence.Column;
-import javax.persistence.Entity;
-import javax.persistence.GeneratedValue;
-import javax.persistence.Id;
+import javax.persistence.MappedSuperclass;
 import java.util.Objects;
 
-@Entity
+@MappedSuperclass
 @Getter
 @NoArgsConstructor(access = AccessLevel.PROTECTED)
-@AllArgsConstructor(access = AccessLevel.PROTECTED)
-public class Secret {
-
-    @Id
-    @GeneratedValue(generator = "system-uuid")
-    @GenericGenerator(name = "system-uuid", strategy = "uuid")
-    private String id;
-
-    private String address; //0xdataset1, aws.amazon.com, beneficiary.key.iex.ec (Kb)
+public abstract class Secret {
     @Column(columnDefinition = "LONGTEXT")
     private String value;
     private boolean isEncryptedValue;
 
     /* Clear secrets at construction */
-    public Secret(String address, String value, boolean isEncryptedValue) {
-        this.address = address;
+    public Secret(String value, boolean isEncryptedValue) {
```

Comment:
The class is now `abstract` and constructor visibility can be reduced
```suggestion
    protected Secret(String value, boolean isEncryptedValue) {
```

Reply:
Please see https://github.com/iExecBlockchainComputing/iexec-sms/pull/140/commits/2895c6ef793446e7c7da492a6b87bc170f7c4c80.